### PR TITLE
Add libzypp callbacks

### PR DESCRIPTION
**Pull request by imobachgs**
## Problem

When something goes wrong during packages installation, the user is not informed.

## Solution

Implement a few `libzypp` callbacks to inform the user about potential problems.

## Testing

- *Added a new unit test*
- *Tested manually*

## To do

- [x] Implement AcceptUnsignedFile (needed for testing)
- [x] Implement MediaChange (to detect whether a package is not in the repository anymore)

### CODE REVIEW

File: service/test/dinstaller/software/callbacks/signature_test.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      it "includes the name and the URL in the question" do
+        expect(subject).to receive(:ask) do |question|
+          expect(question.text).to include("OSS (http://localhost/repo)")
+        end
+        expect(subject.accept_unsigned_file("repomd.xml", 1))
+      end
+    end
+
+    context "when the repo information is not available"
+    it "includes a generic message containing " do
```
Comment (dgdavid) [inline service/test/dinstaller/software/callbacks/signature_test.rb:75]: NP: looks like a not finished sentence.

**Review by dgdavid — APPROVED** [review]
LGTM, just a minor comment.

File: service/lib/dinstaller/software/callbacks/media.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        # @return [String]
+        # @see https://github.com/yast/yast-yast2/blob/19180445ab935a25edd4ae0243aa7a3bcd09c9de/library/packages/src/modules/PackageCallbacks.rb#L620
+        # rubocop:disable Metrics/ParameterLists
+        def media_change(_error_code, error, _url, _product, _current, _current_label, _wanted,
+          _wanted_label, _double_sided, _devices, _current_device)
+          question = DInstaller::Question.new(
+            error, options: [:Retry, :Skip], default_option: :Retry
+          )
+          ask(question) do |q|
+            logger.info "#{q.text}: #{q.answer}"
```
Comment (joseivanlopez) [inline service/lib/dinstaller/software/callbacks/media.rb:64]: NP: Maybe `#ask` should always log this.

File: service/lib/dinstaller/software/manager.rb [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+require "yast2/arch_filter"
+require "dinstaller/software/callbacks"
+
+Yast.import "Package"
+Yast.import "Packages"
+Yast.import "PackageCallbacks"
+Yast.import "Pkg"
+Yast.import "Stage"
+
+# YaST specific code lives under this namespace
```
Comment (joseivanlopez) [inline service/lib/dinstaller/software/manager.rb:36]: What does this mean?

File: service/lib/dinstaller/with_progress.rb [hunk 4]
```diff
@@ -44,6 +47,13 @@ def start_progress(total_steps)
       end
     end
 
+    # Finish the current progress
```
Comment (joseivanlopez) [inline service/lib/dinstaller/with_progress.rb:50]: NP: Finishes

**Review by joseivanlopez — APPROVED** [review]
LGTM

File: service/lib/dinstaller/software/callbacks/media.rb (hunk 2 above)
Comment (imobachgs) [reply service/lib/dinstaller/software/callbacks/media.rb:64]: You are right. I will make a note to change that in another PR (we do the same in the storage callbacks).

File: service/lib/dinstaller/software/manager.rb (hunk 3 above)
Comment (imobachgs) [reply service/lib/dinstaller/software/manager.rb:36]: That's an old comment (5718690aa), but it looks wrong. I will get rid of it.

**Review by joseivanlopez — APPROVED** [review]

**Pull request merged**